File: grpc/hello-world/golang/src/pkg/formula/formula.go
package formula

import (
	"context"
	"log"
	"os"
	"time"

	pb "github.com/GuillaumeFalourd/poc-grpc-golang/helloworld"

	"google.golang.org/grpc"
)

const (
	address     = "localhost:3009"
	defaultName = "world"
)

type Formula struct {
	Text string
}

func (f Formula) Run() {
	// Set up a connection to the server.
	conn, err := grpc.Dial(address, grpc.WithInsecure(), grpc.WithBlock())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	defer conn.Close()
	c := pb.NewGreeterClient(conn)

	// Contact the server and print out its response.
	name := f.Text
	if len(os.Args) > 1 {
		name = os.Args[1]
	}
	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
	defer cancel()

	// Call SayHello method through gRPC.
	r, err := c.SayHello(ctx, &pb.HelloRequest{Name: name})
	if err != nil {
		log.Fatalf("could not greet: %v", err)
	}
	log.Printf("Greeting: %s", r.GetMessage())

	// Call SayHelloAgain method through gRPC.
	r, err = c.SayHelloAgain(ctx, &pb.HelloRequest{Name: name})
	if err != nil {
		log.Fatalf("could not greet: %v", err)
	}
	log.Printf("Greeting: %s", r.GetMessage())
}

File: grpc/hello-world/python/src/formula/formula.py
#!/usr/bin/python3
import logging
import grpc
import os
from datetime import datetime
from formula import message_pb2
from formula import message_pb2_grpc

def run(text, name, surname):
    logging.basicConfig()
    client_request_date = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
    print(f"{client_request_date} - Client Request: Text = \"{text}\", Name = \"{name}\", Surname = \"{surname}\"")

    # Configure channel
    channel = grpc.insecure_channel('localhost:50051')
    stub = message_pb2_grpc.MessageStub(channel)

    # Call the Message method from the message.proto file
    response = stub.Message(message_pb2.MessageRequest(text=text, name=name, surname=surname))
    server_response_date = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
    print(f"{server_response_date} - Server Response: {response.message}")
